locked-asset/simple-lock-energy/src/token_merging.rs

```diff
@@ -0,0 +1,144 @@
+elrond_wasm::imports!();
+
+use simple_lock::locked_token::LockedTokenAttributes;
+
+use crate::energy::Energy;
+
+static CANNOT_MERGE_ERR_MSG: &[u8] = b"Cannot merge";
+
+pub trait Mergeable {
+    fn can_merge_with(&self, other: &Self) -> bool;
+
+    fn merge_with(&mut self, other: Self);
+}
+
+impl<M: ManagedTypeApi> Mergeable for EsdtTokenPayment<M> {
+    fn can_merge_with(&self, other: &Self) -> bool {
```

Comment:
Why assign variables and not just compare the 2 values?

self.token_identifier == other.token_identifier && self.token_nonce == other.token_nonce